src/GogeDao.sol

```diff
@@ -370,718 +476,955 @@ contract GogeDAO is Ownable {
+    /// @param  _addr new governance token address.
+    function _changeGovernanceToken(address _addr) internal {
+        governanceTokenAddr = _addr;
     }
 
-    // ---------- Mutative -----------
+    // ----
+    // View
+    // ----
 
-    function _updateVetoEnabled(bool enabled) internal {
-        require(veto != enabled, "Already set");
-        veto = enabled;
-        emit VetoEnabledUpdated(enabled);
+    /// @notice A view method for returning the amount of votes of a voter in a poll.
+    /// @param  _addr address of voter.
+    /// @param _pollNum unique poll identifier.
+    function getVotes(address _addr, uint256 _pollNum) public view returns (uint256) {
+        return polls[_pollNum][_addr];
     }
 
-    function updateVetoEnabled(bool enabled) external onlyOwner() {
-        _updateVetoEnabled(enabled);
+    /// @notice A view method for returning a poll's unique metadata.
+    /// @param _pollNum unique poll identifier.
+    function getMetadata(uint256 _pollNum) public view returns (Metadata memory) {
+        return pollMap[_pollNum];
     }
 
-    function setTeamMember(address addr, bool value) internal {
-        if(value) {
-            (bool _isTeamMember, ) = isTeamMember(addr);
-            if(!_isTeamMember) teamMembers.push(addr);        
-        } else {
-            (bool _isTeamMember, uint8 s) = isTeamMember(addr);
-            if(_isTeamMember){
-                teamMembers[s] = teamMembers[teamMembers.length - 1];
-                teamMembers.pop();
-            } 
+    /// @notice A view method for returning a poll's current proportion of votes over circuating supply.
+    /// @param _pollNum unique poll identifier.
+    function getProportion(uint256 _pollNum) public view returns (uint256) {
+        return totalVotes[_pollNum] * 100 / IGogeERC20(governanceTokenAddr).getCirculatingMinusReserve();
+    }
+
+    /// @notice A view method for returning the current polls (by pollNum) are active, but expired.
+    /// @return expired array of pollNums that are active, but expired.
+    /// @return counter the amount of polls that are active, but expired.
+    function findExpiredPolls() public view returns (uint256[] memory expired, uint256 counter) {
+        uint256 l = activePolls.length;
+        expired = new uint256[](l);
+
+        for (uint256 i = 0; i < l; i++) {
+            uint256 endTime = pollEndTime[activePolls[i]];
+
+            if (block.timestamp >= endTime) {
+                expired[counter++] = activePolls[i];
+            }
         }
     }
 
-    function setVetoAuthority(address addr, bool value) internal {
-        require(vetoAuthority[addr] != value, "Already set");
-        vetoAuthority[addr] = value;
+    /// @notice A view method for returning whether a given poll is active.
+    /// @param  _pollNum unique poll identifier.
+    /// @return active whether poll is currently active.
+    function isActivePoll(uint256 _pollNum) public view returns (bool active) {
```

Comment:
Also voiced my opinion on this in various functions that use it. Why iterate through potentially all active polls instead of checking if block.timestamp is between the poll's start and end times?